api/tests/Api/ResetPassword/UpdatePasswordTest.php

```diff
@@ -0,0 +1,112 @@
+    // TODO tests for recaptcha check
+    // TODO tests for requesting a password reset and retrieving the password reset data, in separate test files
+
+    protected ?User $user = null;
+    protected ?string $passwordResetKey = null;
+    protected $client;
+
+    public function setUp(): void {
+        parent::setUp();
+        $this->client = static::createBasicClient();
+        $this->client->disableReboot();
+
+        $this->user = static::$fixtures['user1manager'];
+        $this->passwordResetKey = 'dGVzdEBleGFtcGxlLmNvbSM3OWIwZGVkNmEzNGRmNzJkMTU4MzEzNzFlNGVjZWM1ZGYwMWU0ZTc5YzM3ODg4N2IzYjAzOTQzNWNmMmM0MWFj';
+        $this->getEntityManager()->createQueryBuilder()->update(User::class, 'u')
+            ->set('u.passwordResetKeyHash', ':hash')
+            ->where('u.id = :id')
+            ->setParameter('hash', '$2y$13$QjJWNEV/CM1Urnx2kCdJF.Fxj6dFmELvVSxcXkjuzpKYYZYFRmS9q')
+            ->setParameter('id', $this->user->getId())
+            ->getQuery()->execute();
+    }
+
+    public function testPatchResetPasswordValidatesBlankPassword() {
+        $this->mockRecaptcha();
+        $this->client->request('PATCH', '/auth/reset_password/'.$this->passwordResetKey, ['json' => [
+            'password' => '',
+        ], 'headers' => ['Content-Type' => 'application/merge-patch+json']]);
+
+        $this->assertResponseStatusCodeSame(422);
+        $this->assertJsonContains([
+            'violations' => [
+                [
+                    'propertyPath' => 'password',
+                    'message' => 'This value is too short. It should have 12 characters or more.',
+                ],
+            ],
+        ]);
+    }
+
+    public function testPatchResetPasswordValidatesShortPassword() {
+        $this->mockRecaptcha();
+        $this->client->request('PATCH', '/auth/reset_password/'.$this->passwordResetKey, ['json' => [
+            'password' => 'only11chars',
+        ], 'headers' => ['Content-Type' => 'application/merge-patch+json']]);
+
+        $this->assertResponseStatusCodeSame(422);
+        $this->assertJsonContains([
+            'violations' => [
+                [
+                    'propertyPath' => 'password',
+                    'message' => 'This value is too short. It should have 12 characters or more.',
+                ],
+            ],
+        ]);
+    }
+
+    public function testPatchResetPasswordAllowsLongPassword() {
+        $this->mockRecaptcha();
+        $this->client->request('PATCH', '/auth/reset_password/'.$this->passwordResetKey, ['json' => [
+            'password' => 'this password has a total of 122 characters. this password has a total of 122 characters. OWASP approves of this password.',
+        ], 'headers' => ['Content-Type' => 'application/merge-patch+json']]);
+
+        $this->assertResponseStatusCodeSame(200);
+    }
+
+    public function testPatchResetPasswordValidatesUnreasonablyLongPassword() {
+        $this->mockRecaptcha();
```

Comment:
why not call this function directly in the setup?

Reply:
Because I anticipate there will be a test which will call it like `$this->mockRecaptcha(false)` in order to simulate a failed reCaptcha check and test what happens then.